src/panels/GlobeBrowsingPanel/hooks.ts

```diff
@@ -0,0 +1,144 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+
+import {
+  Capability,
+  GlobeBrowsingNodes,
+  LayerType,
+  layerTypes,
+  OpenSpaceCapabilities,
+  OpenSpaceGlobeBrowsingNodes,
+  UrlInfo
+} from './types';
+
+/**
+ * Fetches all layers currently loaded on the globe given by `globeIdentifier`grouped by
+ * their respective `LayerType`
+ * @param globeIdentifier The globe to get layers for
+ * @returns An object containing all layers in their respective group: 'ColorLayers',
+ * 'HeightLayers', 'Overlays', 'HeightLayers', 'WaterMasks'
+ */
+export function useAddedLayers(globeIdentifier: string | null) {
+  const [addedLayers, setAddedLayers] = useState<Record<LayerType, string[]>>({
+    ColorLayers: [],
+    NightLayers: [],
+    Overlays: [],
+    HeightLayers: [],
+    WaterMasks: []
+  });
+  const luaApi = useOpenSpaceApi();
+
+  const fetchLayers = useCallback(async () => {
+    if (!globeIdentifier || !luaApi) {
+      return;
+    }
+
+    const results = await Promise.all(
+      layerTypes.map((type) => luaApi.globebrowsing.layers(globeIdentifier, type))
+    );
+
+    const layers = Object.fromEntries(
+      layerTypes.map((layerType, index) => {
+        const layerObj = results[index] ?? {};
+        const layerList = Object.values(layerObj);
+        return [layerType, layerList];
+      })
+    ) as Record<LayerType, string[]>;
+
+    setAddedLayers(layers);
+  }, [luaApi, globeIdentifier]);
+
+  useEffect(() => {
+    fetchLayers();
+  }, [fetchLayers]);
+
+  return { addedLayers, refresh: fetchLayers };
+}
+
+/**
+ * Fetches all renderable that is of type `renderableGlobe`
+ * @returns An object containing an array of identifiers for all globes currently loaded
+ * in the scene and the index for the first globe in the list without an attached WMS
+ * server
```

Comment:
We should probably indent these, the same way we do in the main repo?

E.g., here: 

```
 * @returns An object containing an array of identifiers for all globes currently loaded
 *          in the scene and the index for the first globe in the list without an attached 
 *          WMS server
```

The same for parameters. Please go through your comments in the PR if we decide we should do this. 

I know it might seem nitpicky, but small things like this do increase readability